Alistairs hacks

ab604 opened the pull request:
I've fixed it as far as making it render html, pdf and epub, but there are still bugs to fix in terms of missing cross-references and labels, 
and I haven't fixed the table in 13 or the figure label in 18. Some of the formatting may need work to make it perfect in the pdf, such as the headers.

Because the errors are so obtuse in Latex, I'm not sure what was the original problem in the end, but these are the things I changed, and the tables seem to be the main source of bugs. I think certain characters such as underscores or brackets may be misinterpreted during rendering.

+ I added some stuff to the Latex preamble to format the pdf.
+ updated the YAML to include the latex preamble and name the rename markdown file for knit-then-merge, `book_filename: Evaluating-What-Works.md`
+ I tweake the index file, but those changes can probably be ignored and it will still work.
+ Change folder to `Data` not `data` on line 62 12-RCR.Rmd
+ 15-Adaptive.Rmd line 77 replaced `adaptive_designs.gif` with `adaptive_designs.png`
+ 13-Drawbacks.Rmd dt13 not rendering `steps-porz` due some character possibly the brackets being misinterpreted by Latex, so created dummy table for now. 
+ 18-Single.Rmd figure label causing problems - again possibly the brackets, created a dummy label.
+ 18-Single.Rmd `readr::read_csv` line 190 and removed underscores from column one name.

It should build from either the build button or using the render_book() with the output defined.

The files I rendered are in the `_book` folder

ab604 committed:
My hacks

ab604 committed:
add missing files and remove old ones

oscci commented:
Thanks so much! I greatly appreciate your efforts. 
I've been through and manually incorporated changes, as I was worried I might screw things up if I pulled them in - I'd pushed updated files to Github and was warned of conflicts and I don't really know what I am doing.
I've changed those issues in tables - and the weird file labelled as .gif instead of .png.
Re the probelmatic table in ch 13 - yesterday I'd downloaded TexStudio, which was v useful in indicating where problems were and it did flag up a line relating to that table. I then realised the table was wrong anyhow, because the original published version had some rows spanning 2 columns. So for now I have saved the original as a figure and used that instead. I'm not even sure it's a very helpful figure, and it's hard to read, but we'll consider its fate in due course - I'm rather reluctant to spend time on it.

I've fixed the .yml and preamble with your changes. But now when I try to render a pdf I get this error:
tlmgr: Local TeX Live (2022) is older than remote repository (2023).
Cross release updates are only supported with
  update-tlmgr-latest(.sh/.exe) --update
See https://tug.org/texlive/upgrade.html for details.
! LaTeX Error: File `fontaxes.sty' not found.

I did try following the link re upgrade, but it said I should not upgrade until a day or two after 19 march!  In fact, I had *yesterday* updated R and all packages, and that included Tex stuff suitable for my Mac, so it's hard to believe it is already out of date. 

My coauthor Paul Thompson understands Latex and has said he can check this out further. 
Meanwhile, Dan Lakens has a very useful tip that I had not known about, which is that people can just issue a print command for any chapter and then either save as pdf or print it out. I think that is a useful workaround meanwhile.

It's reassuring hearing on Twitter that I'm not the only one having these issues - I think I might suggest to Yihue Xie that he adds some warnings about the temperamental nature of pdf rendering - I've spent many hours on this already, and am feeling guilty about involving you too. It might be better if we were not led to believe that pdf was an easy option with bookdown. 

One thing I tried via the TexStudio was to create a pdf from .tex. I suspect that is not too hard, but as it stands it did not know where to find the images.

ab604 commented:
No problem. I didn't intend to get quite so stuck in and although I should have probably been doing chores, after what was a difficult Saturday spending Sunday debugging code was quite a useful distraction.

There's bound to be a clever way, but my basic strategy was:


  1.  Build the index and a single chapter by editing the yml.
  2.  If it failed look at the line error in the output referring to the line in the merged markdown file
  3.  And then identify which bit of the Rmarkdown chapter was causing the problem.
  4.  Comment that out, re-render to check that was the offending bit.
  5.  And then try to fix it.

Regarding the latest error, if that's a Latex package installation error, commenting out the bits of the preamble I added may solve it and it will default to the whatever the base styles are.

From: Dorothy Bishop ***@***.***>
Sent: 20 March 2023 10:21
To: oscci/EvaluatingWhatWorks2023 ***@***.***>
Cc: Alistair Bailey ***@***.***>; Author ***@***.***>
Subject: Re: [oscci/EvaluatingWhatWorks2023] Alistairs hacks (PR #3)

CAUTION: This e-mail originated outside the University of Southampton.

Thanks so much! I greatly appreciate your efforts.
I've been through and manually incorporated changes, as I was worried I might screw things up if I pulled them in - I'd pushed updated files to Github and was warned of conflicts and I don't really know what I am doing.
I've changed those issues in tables - and the weird file labelled as .gif instead of .png.
Re the probelmatic table in ch 13 - yesterday I'd downloaded TexStudio, which was v useful in indicating where problems were and it did flag up a line relating to that table. I then realised the table was wrong anyhow, because the original published version had some rows spanning 2 columns. So for now I have saved the original as a figure and used that instead. I'm not even sure it's a very helpful figure, and it's hard to read, but we'll consider its fate in due course - I'm rather reluctant to spend time on it.

I've fixed the .yml and preamble with your changes. But now when I try to render a pdf I get this error:
tlmgr: Local TeX Live (2022) is older than remote repository (2023).
Cross release updates are only supported with
update-tlmgr-latest(.sh/.exe) --update
See https://tug.org/texlive/upgrade.html<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Ftug.org%2Ftexlive%2Fupgrade.html&data=05%7C01%7Cab604%40soton.ac.uk%7Cf74179c595c047fa321e08db292cd3f5%7C4a5378f929f44d3ebe89669d03ada9d8%7C0%7C0%7C638149044690808673%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=oGavzyDheDzfT1DCGTTCCPWoxg5Z3ztilxyqc2WOlKQ%3D&reserved=0> for details.
! LaTeX Error: File `fontaxes.sty' not found.

I did try following the link re upgrade, but it said I should not upgrade until a day or two after 19 march! In fact, I had yesterday updated R and all packages, and that included Tex stuff suitable for my Mac, so it's hard to believe it is already out of date.

My coauthor Paul Thompson understands Latex and has said he can check this out further.
Meanwhile, Dan Lakens has a very useful tip that I had not known about, which is that people can just issue a print command for any chapter and then either save as pdf or print it out. I think that is a useful workaround meanwhile.

It's reassuring hearing on Twitter that I'm not the only one having these issues - I think I might suggest to Yihue Xie that he adds some warnings about the temperamental nature of pdf rendering - I've spent many hours on this already, and am feeling guilty about involving you too. It might be better if we were not led to believe that pdf was an easy option with bookdown.

One thing I tried via the TexStudio was to create a pdf from .tex. I suspect that is not too hard, but as it stands it did not know where to find the images.

-
Reply to this email directly, view it on GitHub<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Foscci%2FEvaluatingWhatWorks2023%2Fpull%2F3%23issuecomment-1475967040&data=05%7C01%7Cab604%40soton.ac.uk%7Cf74179c595c047fa321e08db292cd3f5%7C4a5378f929f44d3ebe89669d03ada9d8%7C0%7C0%7C638149044690808673%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=zOTchvApGAKBJ%2BGTnHVA8%2Fl2z4wLKoauBj4lorZBglU%3D&reserved=0>, or unsubscribe<https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FACMAS2FJOZ5SAVF6T7YNVQTW5AVRNANCNFSM6AAAAAAWAIUM6A&data=05%7C01%7Cab604%40soton.ac.uk%7Cf74179c595c047fa321e08db292cd3f5%7C4a5378f929f44d3ebe89669d03ada9d8%7C0%7C0%7C638149044690808673%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=Uxa2moTiYtt018EUo84SVoF4lHRFlmLxgy1Q0bkeJvs%3D&reserved=0>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>

oscci commented:
I'm having some success with your strategy. Also, I just got prompted for some updates to one of the Tex packages, and that seems to have got rid of the Local Tex Live problems. 
Many thanks again. I think I'm getting there now.
